### ISSUE TITLE
Server support

### ISSUE THREAD

**Issue by vltr** [techpines/bone.io#16]
Hi! bone.io looks really interesting. I did some work with NodeJS, but I still need to use other server-side programming languages and technologies for other stuff. Mostly I've been using SockJS and it's various implementations on top of other servers.

For now, bone.io seems to work with socket.io and on top of NodeJS (correct me if I'm wrong). Is there a plan for create a "pluggable" system, where I can map bone.io calls to SockJS (just an example) and then make it work with my own technologies? That would be really interesting.

Best regards,
Richard.

**Comment by techpines** [comment]
Yes, I'd like to not have bone.io tied to socket.io.  Socket.io is not even a hard dependency for bone.io right now.  I'd like to keep the API the same, but just provide different adapters for the IO component that can be configured in the `io.options`:

``` js
bone.set('io.options', {
  adapter: 'sockjs',
  ...
});
```

**Comment by vltr** [comment]
That's what I'm talking about :)

I think you shouldn't have to do this work because you'll probably be out of enough adapters to supply general demand :) An "abstract" adapter, perhaps? Or an adapter factory? A good example is the way Ext.Direct works: you can make async calls even if you're server part doesn't fully support streaming :)

**Comment by olituks** [comment]
Hello,
I want to say if soket.io.js is compatible with Google App Engine websoket ?
